lua/zettelkasten/browser.lua

```diff
@@ -1,65 +1,51 @@
 local M = {}
 local fn = vim.fn
 
-local log_levels = vim.log.levels
-local log = require("zettelkasten.log")
 local config = require("zettelkasten.config")
 
 local s_note_cache_with_file_path = {}
 local s_note_cache_with_id = {}
 
-local function get_files(folder, cfg)
+local function get_files(folder, filename_pattern)
     local files = fn.split(fn.globpath(folder, "*.md"), "\\n")
     files = vim.tbl_filter(function(file)
-        return string.match(file, cfg.filename_pattern) ~= nil
+        return string.match(file, filename_pattern) ~= nil
     end, files)
 
     return files
 end
 
-local function extract_id_and_title(line, file_path, cfg)
-    local zk_id = string.match(line, cfg.title_pattern)
+local function extract_id_and_title(
+    line,
+    file_path,
+    title_pattern,
+    id_pattern,
+    id_inference_location
+)
+    local zk_id = string.match(line, title_pattern)
     if zk_id == nil then
         return nil
     end
 
-    local file_id = nil
-    if cfg.id_in_filename then
-        file_id = string.match(vim.fn.fnamemodify(file_path,':t:r'), cfg.id_pattern)
-    end
-
-    zk_id = string.gsub(zk_id, "# ", "")
-    local title_id = nil
     local title = nil
-    if cfg.id_in_title then
-        title_id = string.match(zk_id, cfg.id_pattern)
-        title = vim.trim(string.gsub(zk_id, cfg.id_pattern, ""))
-    else
-        title = vim.trim(zk_id)
-    end
-
     local note_id = nil
-    if cfg.id_in_filename and cfg.id_in_title then
-        if file_id ~= title_id then
-            log.notify('Filename id "'..file_id..'" and title id "'..title_id..'" did not match.', log_levels.ERROR, {})
-            return nil
-        else
-            note_id = file_id
-        end
-    elseif cfg.id_in_filename then
-        note_id = file_id
-    else
-        note_id = title_id
+    zk_id = string.gsub(zk_id, "# ", "")
+    if id_inference_location == 0 then
```

Comment:
Can you assign this `0` and `1` values to some named constants?

Reply:
I don't know if lua can do actual enums. Would using string values suffice?

Reply:
Actually I created confid.TITLE and config.FILENAME and used those. instead of a string.